bfvmm/src/vmcs/test/test_vmcs_intel_x64.cpp

```diff
@@ -20,3 +21,56 @@
 // Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 
 #include <test.h>
+#include <vmcs/vmcs_intel_x64.h>
+#include <memory_manager/memory_manager.h>
+
+static uintptr_t
+virt_to_phys_ptr(void *ptr)
+{
+    (void) ptr;
+
+    return 0x0000000ABCDEF0000;
+}
+
+static void
+setup_vmcs_intrinsics(MockRepository &mocks, memory_manager *mm, intrinsics_intel_x64 *in)
+{
+    // Emulate the memory manager
+    mocks.OnCallFunc(memory_manager::instance).Return(mm);
+    mocks.OnCallOverload(mm, (uintptr_t(memory_manager::*)(void *))&memory_manager::virt_to_phys).Do(virt_to_phys_ptr);
+
+    // Setup MSR and vmread returns to mock a successful vmcs_intel_x64::filter_unsupported
+    mocks.OnCall(in, intrinsics_intel_x64::read_msr).With(IA32_VMX_BASIC_MSR).Return(0x7fffFFFF);
```

Comment:
read / write MSR should call a read_msr / write_msr function with a switch statement that can pull from a std::map which you can prime with bad data if you need to. See here for example:

https://github.com/Bareflank/hypervisor/blob/master/bfvmm/src/serial/test/test_serial_port_intel_x64.cpp